src/citrine/ara/columns.py

```diff
@@ -0,0 +1,83 @@
+"""Column definitions for Ara."""
+from typing import Type, Optional, List  # noqa: F401
+from abc import abstractmethod
+
+from citrine._serialization.serializable import Serializable
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization import properties
+
+
+class Column(PolymorphicSerializable['Column']):
+    """A column in the Ara table, defined as some operation on a variable.
+
+    Abstract type that returns the proper type given a serialized dict.
+    """
+
+    @abstractmethod
+    def _attrs(self) -> List[str]:
+        pass  # pragma: no cover
+
+    def __eq__(self, other):
+        try:
+            return all([
+                self.__getattribute__(key) == other.__getattribute__(key) for key in self._attrs()
+            ])
+        except AttributeError:
+            return False
+
+    @classmethod
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        types: List[Type[Serializable]] = [
+            RealMeanColumn, IdentityColumn
+        ]
+        return next(x for x in types if x.type == data["type"])
+
+
+class RealMeanColumn(Serializable['RealMeanColumn'], Column):
+    """Column containing the mean of a real-valued variable.
+
+    Parameters
+    ----------
+    data_source: str
+        short_name of the variable to use when populating the column
```

Comment:
As per the Slack conversation, `alias` might be more intuitive.  `nickname` operates similarly, but feels less formal.  Would there be value to vetting that there are not collisions in `short_name`s?

Reply:
For the argument name in the column definition?  We had been discussing changing `short_name` in the variable definition.  Do you think `data_source` is confusing?

Reply:
`data_source` is a good concept and good term.